tests/infrastructure/prometheus.go

```diff
@@ -655,3 +599,38 @@ var _ = DescribeInfra("[rfe_id:3187][crit:medium][vendor:cnv-qe@redhat.com][leve
 		Entry("[test_id:6245] by IPv6", k8sv1.IPv6Protocol),
 	)
 })
+
+func countReadyAndLeaderPods(pod *k8sv1.Pod, component string) (foundMetrics map[string]int) {
+	virtClient := kubevirt.Client()
+	target := fmt.Sprintf("virt-%s", component)
+	endpoint, err := virtClient.CoreV1().Endpoints(flags.KubeVirtInstallNamespace).Get(context.Background(), "kubevirt-prometheus-metrics", metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
There are two options to handle errors in helpers:
- Use an assertion with an offset.
- Return the error and let the caller assert.

I always prefer the 2nd option, because assuming the offset is many times impossible.
There are many who prefer the 1st option because it requires less code.
If you keep the assertion, please add the offset.